source/extensions/retry/host/omit_host_metadata/omit_host_metadata.cc

```diff
@@ -0,0 +1,25 @@
+#include "extensions/retry/host/omit_host_metadata/omit_host_metadata.h"
+
+#include "common/config/metadata.h"
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+
+bool OmitHostsRetryPredicate::shouldSelectAnotherHost(const Upstream::Host& host) {
+  // Check if host doesn't have any metadata.
```

Comment:
is this case not handled by `metadataLabelMatch`?

Reply:
Yes, it is indeed. Removed the redundant check.